frontend/src/Hooks/useEditBike.ts

```diff
@@ -0,0 +1,82 @@
+import {ChangeEvent, useState} from "react";
+import {Component} from "../model/Component";
+import axios from "axios";
+import {useNavigate} from "react-router-dom";
+import {ServiceEvent} from "../model/ServiceEvent";
+import {Bike} from "../model/Bike";
+
+type UseEditBikeProps = {
+    editMode: boolean
+    bikes: Bike[]
+    bikeToEdit?: Bike
+    updateBikeList(bikes: Bike[]): void
+}
+export default function useEditBike(props: UseEditBikeProps){
+    const[modelName, setModelName] =
+        useState(props.bikeToEdit? props.bikeToEdit.modelName : "")
+    const[mileage, setMileage] =
+        useState<number | undefined>(props.bikeToEdit? props.bikeToEdit.mileage : undefined)
+    const[mileageFieldValue, setMileageFieldValue] =
+        useState(props.bikeToEdit? props.bikeToEdit.mileage.toString() : "")
+    const[components, setComponents] =
+        useState<Component[]>(props.bikeToEdit? props.bikeToEdit.components : [])
+    const[services, setServices] = useState<ServiceEvent[]>(props.bikeToEdit? props.bikeToEdit.services : [])
+
+    const navigate = useNavigate()
+
+    function handleInputModelName(event: ChangeEvent<HTMLInputElement>){
```

Comment:
You could think about using a reducer for the form state, would maybe keep your code a bit tidier :)
https://medium.com/swlh/usereducer-form-example-16675fa60229